geometry/r3_element_body.hpp

```diff
@@ -180,69 +190,118 @@ R3Element<Scalar> operator-(R3Element<Scalar> const& right) {
 template<typename Scalar>
 R3Element<Scalar> operator+(R3Element<Scalar> const& left,
                             R3Element<Scalar> const& right) {
+#if PRINCIPIA_USE_SSE2_INTRINSICS
+  return R3Element<Scalar>(_mm_add_pd(left.xy, right.xy),
+                           _mm_add_sd(left.zt, right.zt));
+#else
   return R3Element<Scalar>(left.x + right.x,
                            left.y + right.y,
                            left.z + right.z);
+#endif
 }
 
 template<typename Scalar>
 R3Element<Scalar> operator-(R3Element<Scalar> const& left,
                             R3Element<Scalar> const& right) {
+#if PRINCIPIA_USE_SSE2_INTRINSICS
+  return R3Element<Scalar>(_mm_sub_pd(left.xy, right.xy),
+                           _mm_sub_sd(left.zt, right.zt));
+#else
   return R3Element<Scalar>(left.x - right.x,
                            left.y - right.y,
                            left.z - right.z);
+#endif
 }
 
 template<typename Scalar>
 R3Element<Scalar> operator*(double const left,
                             R3Element<Scalar> const& right) {
+#if PRINCIPIA_USE_SSE2_INTRINSICS
+  __m128d const left_128d = ToM128D(left);
+  return R3Element<Scalar>(_mm_mul_pd(left_128d, right.xy),
+                           _mm_mul_sd(left_128d, right.zt));
```

Comment:
It seems nicer to do
```cc
_mm_mul_sd(right.zt, left_128d)
```
for the second one, so that the result has `right.t`; in particular, if we wanted to ensure that `t=0` this would preserve the invariant.

Reply:
It may make sense to use `_mm_set_sd(left)` (and a suitable wrapper in the Quantity case) as the operand for the scalar multiplication, so that we are not waiting for the unpack generated from `_mm_set1_pd`.

Reply:
Moved the constant argument to the right, although it doesn't seem to have much effect.  Not changing to `_mm_set_sd` as it burns an extra register.